### DISCUSSION TITLE
Removing ".html" from page route (while maintaining folder structure)

### DISCUSSION THREAD

**Discussion by earljman** [vuepress/core#882] [General]
This question has been tackled by a couple people VuePress 1.x, but I just wanted check with the community to see if there's been any new progress.

Currently, VuePress generates routes that end in ".html". Is it possible to get rid of this programmatically, without having to restructure folder layout?

The most well-known workaround is to rename all Markdown pages to "index.md" and move them into a folder with the URL you want. 

Example & Workaround:
- The file `src/title.md` creates a page at `localhost:8080/title.html`
- The file `src/title/index.md` creates a page at `localhost:8080/title/`

This creates substantial friction and clutters up the project structure. 

## Potential Hints for a Solution
- Navigating to `localhost:8080/title` _does work_. It just reroutes to `localhost:8080/title.html`. 
- VuePress currently generates paths that contain ".html". This seems like something that could be easily changed via config (if it was supported).
- Perhaps a plugin could be made by modifying the `router` in `clientAppEnhance` ? ([clientAppEnhancedocs](https://v2.vuepress.vuejs.org/advanced/cookbook/usage-of-client-app-enhance.html#use-router-methods))  
   - There used to be a plugin to do this on VuePress 1.x, but [it has been archived](https://github.com/vuepress/vuepress-plugin-clean-urls) for some reason.
- Netlify even has a setting called Pretty URLs that _might_ be helpful. [But apparently, the documentation is not the best](https://answers.netlify.com/t/pretty-urls-url-routing-and-trailing-slashes/29261).

It feels like it this is a valuable feature for the VuePress community, just because it seems to be the obvious preference for most users. Like I mentioned, there were solutions for this before, but I don't see anything for the new VuePress 2.x. Has anybody found anything on this?

**Comment by Mister-Hope** [comment]
You should know that your file will be outputed as `xxx.html` or `index.html`, and `/` means requesting `/index.html` in every static site hosts.

But, we have no garentee that all providers (or all hosters) are redirecting `xxx` to `xxx.html`. Some providers may have stricter settings, which means:

While `/a/b.html` is available, `/A/b.html`, `/a/b` `/a/b/` may get 404, so VuePress currently is providing a safer result, and is reasonable to be the default behavior.

But, if you do want all pages do not have `.html` suffix, you may try to implement one yourself, but I personally do not recommand you to do that.

You can use clientAppEnhance to config router to preform a `xxx.html` to `xxx` redirection if you really wanted to.

**Reply by earljman** [reply]
thanks for the info! Why do you recommend avoiding that implementation?
